ngraph/core/include/ngraph/op/ctc_greedy_decoder.hpp

```diff
@@ -45,10 +45,13 @@ namespace ngraph
                     clone_with_new_inputs(const OutputVector& new_args) const override;
 
                 bool get_ctc_merge_repeated() const { return m_ctc_merge_repeated; }
+                bool evaluate(const HostTensorVector& outputs,
```

Comment:
I am not sure that we can face `CTCGreedyDecoder` on the shape propagation path. 
@lazarevevgeny Do we need evaluate for `CTCGreedyDecoder`?

Reply:
No we don't. Let's move the implementation to tests.

Reply:
@ilyachur can't we just exclude this evaluate() in release build via macros (e.g. #ifndef NDEBUG) ? Then we would have it in debug build for test purposes and it will go away in release build to save space. It will also be easi to switch in in release build in the future when needed.

Reply:
@ilyachur Can you tell me if and what value it brings to implement a reference implementation that is not being executed? Is /will it be used some other way? I`m trying to understand the need of implementing this in the shape it is in now.

Reply:
@lazarevevgeny what you mean by 'move the implementation to tests" ? Where exactly should we put it and how it will be used? We don't understand how it will be useful if there is no evaluate().

Reply:
@lazarevevgeny please take a look if the current implementation matches what you've proposed.

Reply:
We had a meeting to discuss this - all is clear now. For current moment implementation should go to int_executable and we should not have evaluate() method. In this way it will be available in INTERPRETER but not present in final production binary.